--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftTransmuteRecipe.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftTransmuteRecipe.java
@@ -4,15 +_,15 @@
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.world.item.crafting.RecipeHolder;
 import net.minecraft.world.item.crafting.TransmuteResult;
-import org.bukkit.Material;
 import org.bukkit.NamespacedKey;
 import org.bukkit.craftbukkit.util.CraftNamespacedKey;
+import org.bukkit.inventory.ItemType;
 import org.bukkit.inventory.RecipeChoice;
 import org.bukkit.inventory.TransmuteRecipe;
 
 public class CraftTransmuteRecipe extends TransmuteRecipe implements CraftRecipe {
 
-    public CraftTransmuteRecipe(NamespacedKey key, Material result, RecipeChoice input, RecipeChoice material) {
+    public CraftTransmuteRecipe(NamespacedKey key, ItemType result, RecipeChoice input, RecipeChoice material) {
         super(key, result, input, material);
     }
 
