--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftInventory.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftInventory.java
@@ -24,13 +_,12 @@
 import net.minecraft.world.level.block.entity.ShulkerBoxBlockEntity;
 import net.minecraft.world.level.block.entity.SmokerBlockEntity;
 import org.bukkit.Location;
-import org.bukkit.Material;
-import org.bukkit.craftbukkit.util.CraftLegacy;
 import org.bukkit.entity.HumanEntity;
 import org.bukkit.event.inventory.InventoryType;
 import org.bukkit.inventory.Inventory;
 import org.bukkit.inventory.InventoryHolder;
 import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.ItemType;
 
 public class CraftInventory implements Inventory {
     protected final Container inventory;
@@ -102,9 +_,8 @@
     }
 
     @Override
-    public boolean contains(Material material) {
+    public boolean contains(ItemType material) {
         Preconditions.checkArgument(material != null, "Material cannot be null");
-        material = CraftLegacy.fromLegacy(material);
         for (ItemStack item : this.getStorageContents()) {
             if (item != null && item.getType() == material) {
                 return true;
@@ -127,9 +_,8 @@
     }
 
     @Override
-    public boolean contains(Material material, int amount) {
+    public boolean contains(ItemType material, int amount) {
         Preconditions.checkArgument(material != null, "Material cannot be null");
-        material = CraftLegacy.fromLegacy(material);
         if (amount <= 0) {
             return true;
         }
@@ -176,9 +_,8 @@
     }
 
     @Override
-    public HashMap<Integer, ItemStack> all(Material material) {
+    public HashMap<Integer, ItemStack> all(ItemType material) {
         Preconditions.checkArgument(material != null, "Material cannot be null");
-        material = CraftLegacy.fromLegacy(material);
         HashMap<Integer, ItemStack> slots = new HashMap<>();
 
         ItemStack[] inventory = this.getStorageContents();
@@ -206,9 +_,8 @@
     }
 
     @Override
-    public int first(Material material) {
+    public int first(ItemType material) {
         Preconditions.checkArgument(material != null, "Material cannot be null");
-        material = CraftLegacy.fromLegacy(material);
         ItemStack[] inventory = this.getStorageContents();
         for (int i = 0; i < inventory.length; i++) {
             ItemStack item = inventory[i];
@@ -405,9 +_,8 @@
     }
 
     @Override
-    public void remove(Material material) {
+    public void remove(ItemType material) {
         Preconditions.checkArgument(material != null, "Material cannot be null");
-        material = CraftLegacy.fromLegacy(material);
         ItemStack[] items = this.getStorageContents();
         for (int i = 0; i < items.length; i++) {
             if (items[i] != null && items[i].getType() == material) {
