--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaEntityTag.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaEntityTag.java
@@ -11,21 +_,21 @@
 import net.minecraft.nbt.Tag;
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.item.component.TypedEntityData;
-import org.bukkit.Material;
 import org.bukkit.configuration.serialization.DelegateDeserialization;
+import org.bukkit.inventory.ItemType;
 
 @DelegateDeserialization(SerializableMeta.class)
 public class CraftMetaEntityTag extends CraftMetaItem {
 
-    private static final Set<Material> ENTITY_TAGGABLE_MATERIALS = Sets.newHashSet(
-            Material.COD_BUCKET,
-            Material.PUFFERFISH_BUCKET,
-            Material.TADPOLE_BUCKET,
-            Material.SALMON_BUCKET,
-            Material.SULFUR_CUBE_BUCKET,
-            Material.ITEM_FRAME,
-            Material.GLOW_ITEM_FRAME,
-            Material.PAINTING
+    private static final Set<ItemType> ENTITY_TAGGABLE_MATERIALS = Sets.newHashSet(
+            ItemType.COD_BUCKET,
+            ItemType.PUFFERFISH_BUCKET,
+            ItemType.TADPOLE_BUCKET,
+            ItemType.SALMON_BUCKET,
+            ItemType.SULFUR_CUBE_BUCKET,
+            ItemType.ITEM_FRAME,
+            ItemType.GLOW_ITEM_FRAME,
+            ItemType.PAINTING
     );
 
     static final ItemMetaKeyType<TypedEntityData<EntityType<?>>> ENTITY_TAG = new ItemMetaKeyType<>(DataComponents.ENTITY_DATA, "EntityTag", "entity-tag");
@@ -77,7 +_,7 @@
     }
 
     @Override
-    boolean applicableTo(Material type) {
+    boolean applicableTo(ItemType type) {
         return CraftMetaEntityTag.ENTITY_TAGGABLE_MATERIALS.contains(type);
     }
 
