--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemType.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemType.java
@@ -28,7 +_,6 @@
 import net.minecraft.world.level.storage.loot.providers.number.ints.NumberDispatcher;
 import net.minecraft.world.level.storage.loot.providers.number.ints.ResolvableInt;
 import net.minecraft.world.level.storage.loot.providers.number.ints.WeightedListValue;
-import org.bukkit.Material;
 import org.bukkit.Registry;
 import org.bukkit.World;
 import org.bukkit.attribute.Attribute;
@@ -54,14 +_,6 @@
 
     private final Supplier<CraftItemMetas.ItemMetaData<M>> itemMetaData;
 
-    public static Material minecraftToBukkit(Item minecraft) {
-        return CraftMagicNumbers.getMaterial(minecraft);
-    }
-
-    public static Item bukkitToMinecraft(Material bukkit) {
-        return CraftMagicNumbers.getItem(bukkit);
-    }
-
     public static ItemType minecraftToBukkitNew(Item minecraft) {
         return CraftRegistry.minecraftToBukkit(minecraft, Registries.ITEM);
     }
@@ -284,11 +_,6 @@
     @Override
     public String getTranslationKey() {
         return this.getHandle().getDescriptionId();
-    }
-
-    @Override
-    public @Nullable Material asMaterial() {
-        return Registry.MATERIAL.get(this.getKey());
     }
 
     // Paper start - add Translatable
