--- a/src/main/java/org/bukkit/entity/Animals.java
+++ b/src/main/java/org/bukkit/entity/Animals.java
@@ -1,8 +_,8 @@
 package org.bukkit.entity;
 
 import java.util.UUID;
-import org.bukkit.Material;
 import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.ItemType;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
@@ -65,12 +_,12 @@
     boolean isBreedItem(@NotNull ItemStack stack);
 
     /**
-     * Check if the provided Material is the correct item used for breeding
+     * Check if the provided ItemType is the correct item used for breeding
      * this entity.
      *
      * @param material Material to check.
      * @return if the provided Material is the correct food item for this
      * entity.
      */
-    boolean isBreedItem(@NotNull Material material);
+    boolean isBreedItem(@NotNull ItemType material);
 }
