--- a/src/main/java/org/bukkit/entity/Enderman.java
+++ b/src/main/java/org/bukkit/entity/Enderman.java
@@ -1,7 +_,6 @@
 package org.bukkit.entity;
 
 import org.bukkit.block.data.BlockData;
-import org.bukkit.material.MaterialData;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
@@ -20,25 +_,6 @@
 
     public boolean teleportRandomly();
     // Paper end
-
-    /**
-     * Gets the id and data of the block that the Enderman is carrying.
-     *
-     * @return MaterialData containing the id and data of the block
-     * @deprecated use {@link #getCarriedBlock()}
-     */
-    @NotNull
-    @Deprecated(forRemoval = true, since = "1.13")
-    public MaterialData getCarriedMaterial();
-
-    /**
-     * Sets the id and data of the block that the Enderman is carrying.
-     *
-     * @param material data to set the carried block to
-     * @deprecated use {@link #setCarriedBlock(BlockData)}
-     */
-    @Deprecated(forRemoval = true, since = "1.13")
-    public void setCarriedMaterial(@NotNull MaterialData material);
 
     /**
      * Gets the data of the block that the Enderman is carrying.
