--- a/src/main/java/org/bukkit/entity/HumanEntity.java
+++ b/src/main/java/org/bukkit/entity/HumanEntity.java
@@ -10,7 +_,6 @@
 import net.kyori.adventure.key.Key;
 import org.bukkit.GameMode;
 import org.bukkit.Location;
-import org.bukkit.Material;
 import org.bukkit.NamespacedKey;
 import org.bukkit.inventory.EntityEquipment;
 import org.bukkit.inventory.EquipmentSlot;
@@ -18,6 +_,7 @@
 import org.bukkit.inventory.InventoryHolder;
 import org.bukkit.inventory.InventoryView;
 import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.ItemType;
 import org.bukkit.inventory.MainHand;
 import org.bukkit.inventory.MenuType;
 import org.bukkit.inventory.Merchant;
@@ -356,25 +_,23 @@
     public void setItemOnCursor(@Nullable ItemStack item);
 
     /**
-     * Check whether a cooldown is active on the specified material.
+     * Check whether a cooldown is active on the specified item type.
      *
-     * @param material the material to check
-     * @return if a cooldown is active on the material
-     * @throws IllegalArgumentException if the material is not an item
+     * @param material the item type to check
+     * @return if a cooldown is active on the item type
      */
-    public boolean hasCooldown(Material material);
+    public boolean hasCooldown(ItemType material);
 
     /**
-     * Get the cooldown time in ticks remaining for the specified material.
+     * Get the cooldown time in ticks remaining for the specified item type.
      *
-     * @param material the material to check
+     * @param material the item type to check
      * @return the remaining cooldown time in ticks
-     * @throws IllegalArgumentException if the material is not an item
      */
-    public int getCooldown(Material material);
+    public int getCooldown(ItemType material);
 
     /**
-     * Set a cooldown on the specified material for a certain amount of ticks.
+     * Set a cooldown on the specified item type for a certain amount of ticks.
      * 0 ticks will result in the removal of the cooldown.
      * <p>
      * Cooldowns are used by the server for items such as ender pearls and
@@ -383,11 +_,10 @@
      * Note that cooldowns will not by themselves stop an item from being used
      * for attacking.
      *
-     * @param material the material to set the cooldown for
+     * @param material the item type to set the cooldown for
      * @param ticks the amount of ticks to set or 0 to remove
-     * @throws IllegalArgumentException if the material is not an item
      */
-    default void setCooldown(Material material, int ticks) {
+    default void setCooldown(ItemType material, int ticks) {
         this.setCooldown(ItemStack.of(material), ticks);
     }
 
