--- a/src/main/java/org/bukkit/craftbukkit/generator/CraftChunkData.java
+++ b/src/main/java/org/bukkit/craftbukkit/generator/CraftChunkData.java
@@ -10,16 +_,14 @@
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.chunk.ChunkAccess;
 import org.bukkit.HeightMap;
-import org.bukkit.Material;
 import org.bukkit.World;
 import org.bukkit.block.Biome;
+import org.bukkit.block.BlockType;
 import org.bukkit.block.data.BlockData;
 import org.bukkit.craftbukkit.CraftHeightMap;
 import org.bukkit.craftbukkit.block.CraftBiome;
 import org.bukkit.craftbukkit.block.data.CraftBlockData;
-import org.bukkit.craftbukkit.util.CraftMagicNumbers;
 import org.bukkit.generator.ChunkGenerator;
-import org.bukkit.material.MaterialData;
 
 /**
  * Data to be used for the block types and data in a newly generated chunk.
@@ -68,46 +_,31 @@
     }
 
     @Override
-    public void setBlock(int x, int y, int z, Material material) {
+    public void setBlock(int x, int y, int z, BlockType material) {
         this.setBlock(x, y, z, material.createBlockData());
     }
 
     @Override
-    public void setBlock(int x, int y, int z, MaterialData material) {
-        this.setBlock(x, y, z, CraftMagicNumbers.getBlock(material));
-    }
-
-    @Override
     public void setBlock(int x, int y, int z, BlockData blockData) {
         this.setBlock(x, y, z, ((CraftBlockData) blockData).getState());
     }
 
     @Override
-    public void setRegion(int xMin, int yMin, int zMin, int xMax, int yMax, int zMax, Material material) {
+    public void setRegion(int xMin, int yMin, int zMin, int xMax, int yMax, int zMax, BlockType material) {
         this.setRegion(xMin, yMin, zMin, xMax, yMax, zMax, material.createBlockData());
     }
 
     @Override
-    public void setRegion(int xMin, int yMin, int zMin, int xMax, int yMax, int zMax, MaterialData material) {
-        this.setRegion(xMin, yMin, zMin, xMax, yMax, zMax, CraftMagicNumbers.getBlock(material));
-    }
-
-    @Override
     public void setRegion(int xMin, int yMin, int zMin, int xMax, int yMax, int zMax, BlockData blockData) {
         this.setRegion(xMin, yMin, zMin, xMax, yMax, zMax, ((CraftBlockData) blockData).getState());
     }
 
     @Override
-    public Material getType(int x, int y, int z) {
+    public BlockType getType(int x, int y, int z) {
         return this.getTypeId(x, y, z).getBukkitMaterial(); // Paper - optimise getType calls
     }
 
     @Override
-    public MaterialData getTypeAndData(int x, int y, int z) {
-        return CraftMagicNumbers.getMaterial(this.getTypeId(x, y, z));
-    }
-
-    @Override
     public BlockData getBlockData(int x, int y, int z) {
         return this.getTypeId(x, y, z).asBlockData();
     }
@@ -154,11 +_,6 @@
 
         ChunkAccess chunk = this.getHandle();
         return chunk.getBlockState(new BlockPos(chunk.getPos().getMinBlockX() + x, y, chunk.getPos().getMinBlockZ() + z));
-    }
-
-    @Override
-    public byte getData(int x, int y, int z) {
-        return CraftMagicNumbers.toLegacyData(this.getTypeId(x, y, z));
     }
 
     private void setBlock(int x, int y, int z, BlockState state) {
