--- a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
@@ -71,7 +_,6 @@
 import net.minecraft.world.level.storage.TagValueOutput;
 import org.bukkit.Bukkit;
 import org.bukkit.Keyed;
-import org.bukkit.Material;
 import org.bukkit.NamespacedKey;
 import org.bukkit.UnsafeValues;
 import org.bukkit.World;
@@ -83,12 +_,10 @@
 import org.bukkit.craftbukkit.CraftWorld;
 import org.bukkit.craftbukkit.entity.CraftEntity;
 import org.bukkit.craftbukkit.inventory.CraftItemStack;
-import org.bukkit.craftbukkit.legacy.CraftLegacy;
 import org.bukkit.craftbukkit.legacy.FieldRename;
 import org.bukkit.craftbukkit.potion.CraftPotionType;
 import org.bukkit.enchantments.Enchantment;
 import org.bukkit.inventory.ItemStack;
-import org.bukkit.material.MaterialData;
 import org.bukkit.plugin.InvalidPluginException;
 import org.bukkit.plugin.PluginDescriptionFile;
 import org.bukkit.potion.PotionType;
@@ -103,142 +_,8 @@
     public static final CraftMagicNumbers INSTANCE = new CraftMagicNumbers();
     public static final boolean DISABLE_OLD_API_SUPPORT = Boolean.getBoolean("paper.disableOldApiSupport"); // Paper
 
-    private final Commodore commodore = new Commodore();
-
     private CraftMagicNumbers() {}
 
-    public static BlockState getBlock(MaterialData material) {
-        return CraftMagicNumbers.getBlock(material.getItemType(), material.getData());
-    }
-
-    public static BlockState getBlock(Material material, byte data) {
-        return CraftLegacy.fromLegacyData(CraftLegacy.toLegacy(material), data);
-    }
-
-    public static MaterialData getMaterial(BlockState data) {
-        return CraftLegacy.toLegacy(CraftMagicNumbers.getMaterial(data.getBlock())).getNewData(CraftMagicNumbers.toLegacyData(data));
-    }
-
-    public static Item getItem(Material material, short data) {
-        if (material.isLegacy()) {
-            return CraftLegacy.fromLegacyData(CraftLegacy.toLegacy(material), data);
-        }
-
-        return CraftMagicNumbers.getItem(material);
-    }
-
-    public static MaterialData getMaterialData(Item item) {
-        return CraftLegacy.toLegacyData(CraftMagicNumbers.getMaterial(item));
-    }
-
-    // ========================================================================
-    private static final Map<Block, Material> BLOCK_MATERIAL = new IdentityHashMap<>();
-    private static final Map<Item, Material> ITEM_MATERIAL = new IdentityHashMap<>();
-    private static final Map<Material, Item> MATERIAL_ITEM = new EnumMap<>(Material.class);
-    private static final Map<Material, Block> MATERIAL_BLOCK = new EnumMap<>(Material.class);
-
-    static {
-        for (Block block : BuiltInRegistries.BLOCK) {
-            BLOCK_MATERIAL.put(block, Material.getMaterial(BuiltInRegistries.BLOCK.getKey(block).getPath().toUpperCase(Locale.ROOT)));
-        }
-
-        for (Item item : BuiltInRegistries.ITEM) {
-            ITEM_MATERIAL.put(item, Material.getMaterial(BuiltInRegistries.ITEM.getKey(item).getPath().toUpperCase(Locale.ROOT)));
-        }
-
-        for (Material material : Material.values()) {
-            if (material.isLegacy()) {
-                continue;
-            }
-
-            Identifier key = CraftNamespacedKey.toMinecraft(material.getKey());
-            BuiltInRegistries.ITEM.getOptional(key).ifPresent((item) -> {
-                CraftMagicNumbers.MATERIAL_ITEM.put(material, item);
-            });
-            BuiltInRegistries.BLOCK.getOptional(key).ifPresent((block) -> {
-                CraftMagicNumbers.MATERIAL_BLOCK.put(material, block);
-            });
-        }
-    }
-
-    public static Material getMaterial(Block block) {
-        return CraftMagicNumbers.BLOCK_MATERIAL.get(block);
-    }
-
-    public static Material getMaterial(Item item) {
-        return CraftMagicNumbers.ITEM_MATERIAL.getOrDefault(item, Material.AIR);
-    }
-
-    public static Item getItem(Material material) {
-        if (material != null && material.isLegacy()) {
-            material = CraftLegacy.fromLegacy(material);
-        }
-
-        return CraftMagicNumbers.MATERIAL_ITEM.get(material);
-    }
-
-    public static Block getBlock(Material material) {
-        if (material != null && material.isLegacy()) {
-            material = CraftLegacy.fromLegacy(material);
-        }
-
-        return CraftMagicNumbers.MATERIAL_BLOCK.get(material);
-    }
-    // ========================================================================
-
-    public static byte toLegacyData(BlockState data) {
-        return CraftLegacy.toLegacyData(data);
-    }
-
-    public Commodore getCommodore() {
-        return this.commodore;
-    }
-
-    @Override
-    public Material toLegacy(Material material) {
-        return CraftLegacy.toLegacy(material);
-    }
-
-    @Override
-    public Material fromLegacy(Material material) {
-        return CraftLegacy.fromLegacy(material);
-    }
-
-    @Override
-    public Material fromLegacy(MaterialData material) {
-        return CraftLegacy.fromLegacy(material);
-    }
-
-    @Override
-    public Material fromLegacy(MaterialData material, boolean itemPriority) {
-        return CraftLegacy.fromLegacy(material, itemPriority);
-    }
-
-    @Override
-    public BlockData fromLegacy(Material material, byte data) {
-        return CraftMagicNumbers.getBlock(material, data).asBlockData();
-    }
-
-    @Override
-    public Material getMaterial(String material, int version) {
-        Preconditions.checkArgument(material != null, "material == null");
-        Preconditions.checkArgument(version <= this.getDataVersion(), "Newer version! Server downgrades are not supported!");
-
-        // Fastpath up to date materials
-        if (version == this.getDataVersion()) {
-            return Material.getMaterial(material);
-        }
-
-        Dynamic<Tag> name = new Dynamic<>(NbtOps.INSTANCE, StringTag.valueOf("minecraft:" + material.toLowerCase(Locale.ROOT)));
-        Dynamic<Tag> converted = DataFixers.getDataFixer().update(References.ITEM_NAME, name, version, this.getDataVersion());
-
-        if (name.equals(converted)) {
-            converted = DataFixers.getDataFixer().update(References.BLOCK_NAME, name, version, this.getDataVersion());
-        }
-
-        return Material.matchMaterial(converted.asString(""));
-    }
-
     @Override
     public int getDataVersion() {
         return SharedConstants.getCurrentVersion().dataVersion().version();
@@ -383,10 +_,6 @@
             throw new InvalidPluginException("Plugin API version " + descriptionFile.getAPIVersion() + " is lower than the minimum allowed version. Please update or replace it.");
         }
 
-        if (!DISABLE_OLD_API_SUPPORT && toCheck.isOlderThan(ApiVersion.FLATTENING)) { // Paper
-            CraftLegacy.init();
-        }
-
         if (toCheck == ApiVersion.NONE) {
             Bukkit.getLogger().log(Level.WARNING, "Legacy plugin " + descriptionFile.getFullName() + " does not specify an api-version.");
         }
@@ -398,17 +_,6 @@
 
     @Override
     public byte[] processClass(PluginDescriptionFile pdf, String path, byte[] clazz) {
-        // Paper start
-        if (DISABLE_OLD_API_SUPPORT) {
-            return clazz;
-        }
-        // Paper end
-        try {
-            clazz = this.commodore.convert(clazz, pdf.getName(), ApiVersion.getOrCreateVersion(pdf.getAPIVersion()), ((CraftServer) Bukkit.getServer()).activeCompatibilities);
-        } catch (Exception ex) {
-            Bukkit.getLogger().log(Level.SEVERE, "Fatal error trying to convert " + pdf.getFullName() + ":" + path, ex);
-        }
-
         return clazz;
     }
 
