--- a/src/main/java/org/bukkit/event/player/PlayerBucketEvent.java
+++ b/src/main/java/org/bukkit/event/player/PlayerBucketEvent.java
@@ -1,14 +_,12 @@
 package org.bukkit.event.player;
 
-import org.bukkit.Bukkit;
-import org.bukkit.Material;
 import org.bukkit.block.Block;
 import org.bukkit.block.BlockFace;
 import org.bukkit.entity.Player;
 import org.bukkit.event.Cancellable;
 import org.bukkit.inventory.EquipmentSlot;
 import org.bukkit.inventory.ItemStack;
-import org.bukkit.material.MaterialData;
+import org.bukkit.inventory.ItemType;
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
@@ -21,7 +_,7 @@
     private final Block block;
     private final Block blockClicked;
     private final BlockFace blockFace;
-    private final Material bucket;
+    private final ItemType bucket;
     private final EquipmentSlot hand;
     private ItemStack itemStack;
 
@@ -29,28 +_,24 @@
 
     @ApiStatus.Internal
     @Deprecated(since = "1.14.4", forRemoval = true)
-    public PlayerBucketEvent(@NotNull final Player player, @NotNull final Block blockClicked, @NotNull final BlockFace blockFace, @NotNull final Material bucket, @NotNull final ItemStack itemInHand) {
+    public PlayerBucketEvent(@NotNull final Player player, @NotNull final Block blockClicked, @NotNull final BlockFace blockFace, @NotNull final ItemType bucket, @NotNull final ItemStack itemInHand) {
         this(player, null, blockClicked.getRelative(blockFace), blockFace, bucket, itemInHand, EquipmentSlot.HAND);
     }
 
     @ApiStatus.Internal
     @Deprecated(since = "1.19.2", forRemoval = true)
-    public PlayerBucketEvent(@NotNull final Player player, @NotNull final Block block, @NotNull final Block blockClicked, @NotNull final BlockFace blockFace, @NotNull final Material bucket, @NotNull final ItemStack itemInHand) {
+    public PlayerBucketEvent(@NotNull final Player player, @NotNull final Block block, @NotNull final Block blockClicked, @NotNull final BlockFace blockFace, @NotNull final ItemType bucket, @NotNull final ItemStack itemInHand) {
         this(player, block, blockClicked, blockFace, bucket, itemInHand, EquipmentSlot.HAND);
     }
 
     @ApiStatus.Internal
-    public PlayerBucketEvent(@NotNull final Player player, @NotNull final Block block, @NotNull final Block blockClicked, @NotNull final BlockFace blockFace, @NotNull final Material bucket, @NotNull final ItemStack itemInHand, @NotNull final EquipmentSlot hand) {
+    public PlayerBucketEvent(@NotNull final Player player, @NotNull final Block block, @NotNull final Block blockClicked, @NotNull final BlockFace blockFace, @NotNull final ItemType bucket, @NotNull final ItemStack itemInHand, @NotNull final EquipmentSlot hand) {
         super(player);
         this.block = block;
         this.blockClicked = blockClicked;
         this.blockFace = blockFace;
         this.itemStack = itemInHand;
-        if (bucket != null && bucket.isLegacy()) {
-            this.bucket = Bukkit.getUnsafe().fromLegacy(new MaterialData(bucket), true);
-        } else {
-            this.bucket = bucket;
-        }
+        this.bucket = bucket;
         this.hand = hand;
     }
 
@@ -90,7 +_,7 @@
      * @return the used bucket
      */
     @NotNull
-    public Material getBucket() {
+    public ItemType getBucket() {
         return this.bucket;
     }
 
