--- a/src/main/java/org/bukkit/event/player/PlayerShearEntityEvent.java
+++ b/src/main/java/org/bukkit/event/player/PlayerShearEntityEvent.java
@@ -1,12 +_,12 @@
 package org.bukkit.event.player;
 
-import org.bukkit.Material;
 import org.bukkit.entity.Entity;
 import org.bukkit.entity.Player;
 import org.bukkit.event.Cancellable;
 import org.bukkit.event.HandlerList;
 import org.bukkit.inventory.EquipmentSlot;
 import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.ItemType;
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Unmodifiable;
@@ -38,7 +_,7 @@
     @ApiStatus.Internal
     @Deprecated(since = "1.15.2", forRemoval = true)
     public PlayerShearEntityEvent(@NotNull final Player player, @NotNull final Entity entity) {
-        this(player, entity, new ItemStack(Material.SHEARS), EquipmentSlot.HAND, java.util.Collections.emptyList());
+        this(player, entity, new ItemStack(ItemType.SHEARS), EquipmentSlot.HAND, java.util.Collections.emptyList());
     }
 
     /**
