--- a/src/main/java/org/bukkit/event/player/PlayerBucketFishEvent.java
+++ b/src/main/java/org/bukkit/event/player/PlayerBucketFishEvent.java
@@ -1,6 +_,5 @@
 package org.bukkit.event.player;
 
-import org.bukkit.Material;
 import org.bukkit.Warning;
 import org.bukkit.entity.Fish;
 import org.bukkit.entity.Player;
@@ -37,7 +_,7 @@
     /**
      * Gets the bucket used.
      * <br>
-     * This refers to the bucket clicked with, ie {@link Material#WATER_BUCKET}.
+     * This refers to the bucket clicked with, ie {@link org.bukkit.inventory.ItemType#WATER_BUCKET}.
      *
      * @return The used bucket
      * @deprecated Use {@link #getOriginalBucket()}
@@ -52,7 +_,7 @@
      * Gets the bucket that the fish will be put into.
      * <br>
      * This refers to the bucket with the fish, ie
-     * {@link Material#PUFFERFISH_BUCKET}.
+     * {@link org.bukkit.inventory.ItemType#PUFFERFISH_BUCKET}.
      *
      * @return The bucket that the fish will be put into
      * @deprecated Use {@link #getEntityBucket()}
