--- a/src/main/java/org/bukkit/block/data/BlockData.java
+++ b/src/main/java/org/bukkit/block/data/BlockData.java
@@ -2,17 +_,18 @@
 
 import org.bukkit.Color;
 import org.bukkit.Location;
-import org.bukkit.Material;
 import org.bukkit.Server;
 import org.bukkit.SoundGroup;
 import org.bukkit.block.Block;
 import org.bukkit.block.BlockFace;
 import org.bukkit.block.BlockState;
 import org.bukkit.block.BlockSupport;
+import org.bukkit.block.BlockType;
 import org.bukkit.block.PistonMoveReaction;
 import org.bukkit.block.structure.Mirror;
 import org.bukkit.block.structure.StructureRotation;
 import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.ItemType;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
@@ -24,7 +_,7 @@
      * @return the material
      */
     @NotNull
-    Material getMaterial();
+    BlockType getMaterial();
 
     /**
      * Gets a string, which when passed into a method such as
@@ -233,13 +_,13 @@
      *
      * For example:
      * <pre>
-     * {@link Material#REDSTONE_WIRE} -> {@link Material#REDSTONE}
-     * {@link Material#CARROTS} -> {@link Material#CARROT}
+     * {@link BlockType#REDSTONE_WIRE} -> {@link ItemType#REDSTONE}
+     * {@link BlockType#CARROTS} -> {@link ItemType#CARROT}
      * </pre>
-     * @return placement material or {@link Material#AIR} if it doesn't have one
+     * @return placement material or {@link ItemType#AIR} if it doesn't have one
      */
     @NotNull
-    Material getPlacementMaterial();
+    ItemType getPlacementMaterial();
 
     /**
      * Rotates this blockdata by the specified {@link StructureRotation}.
