--- a/src/main/java/org/bukkit/inventory/BlastingRecipe.java
+++ b/src/main/java/org/bukkit/inventory/BlastingRecipe.java
@@ -1,15 +_,14 @@
 package org.bukkit.inventory;
 
-import org.bukkit.Material;
 import org.bukkit.NamespacedKey;
 import org.jetbrains.annotations.NotNull;
 
 /**
- * Represents a campfire recipe.
+ * Represents a blasting recipe.
  */
 public class BlastingRecipe extends CookingRecipe<BlastingRecipe> {
 
-    public BlastingRecipe(@NotNull NamespacedKey key, @NotNull ItemStack result, @NotNull Material source, float experience, int cookingTime) {
+    public BlastingRecipe(@NotNull NamespacedKey key, @NotNull ItemStack result, @NotNull ItemType source, float experience, int cookingTime) {
         super(key, result, source, experience, cookingTime);
     }
 
