--- a/src/main/java/org/bukkit/inventory/ItemType.java
+++ b/src/main/java/org/bukkit/inventory/ItemType.java
@@ -7,7 +_,6 @@
 import net.kyori.adventure.key.Key;
 import net.kyori.adventure.key.KeyPattern;
 import org.bukkit.Keyed;
-import org.bukkit.Material;
 import org.bukkit.Registry;
 import org.bukkit.Translatable;
 import org.bukkit.World;
@@ -3602,15 +_,6 @@
      */
     @Deprecated(forRemoval = true, since = "1.21.1") // Paper
     boolean isEnabledByFeature(World world);
-
-    /**
-     * Tries to convert this ItemType into a Material
-     *
-     * @return the converted Material or null
-     * @deprecated only for internal use
-     */
-    @Deprecated(since = "1.20.6")
-    @Nullable Material asMaterial();
 
     /**
      * @deprecated use {@link #translationKey()} and {@link net.kyori.adventure.text.Component#translatable(net.kyori.adventure.translation.Translatable)}
