--- a/src/main/java/org/bukkit/inventory/RecipeChoice.java
+++ b/src/main/java/org/bukkit/inventory/RecipeChoice.java
@@ -12,10 +_,6 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.function.Predicate;
-import org.bukkit.Bukkit;
-import org.bukkit.Material;
-import org.bukkit.Tag;
-import org.bukkit.material.MaterialData;
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.Contract;
 import org.jspecify.annotations.NullMarked;
@@ -139,135 +_,6 @@
     // Paper end - check valid ingredients
 
     /**
-     * Represents a choice of multiple matching Materials.
-     * @apiNote recommended to use {@link ItemTypeChoice}
-     */
-    @ApiStatus.Obsolete(since = "1.21.11")
-    sealed class MaterialChoice implements RecipeChoice permits ItemTypeRecipeChoiceImpl {
-
-        protected MaterialChoice() {
-        }
-
-        private List<Material> choices;
-
-        public MaterialChoice(Material choice) {
-            this(Arrays.asList(choice));
-        }
-
-        public MaterialChoice(Material... choices) {
-            this(Arrays.asList(choices));
-        }
-
-        /**
-         * Constructs a MaterialChoice with the current values of the specified
-         * tag.
-         *
-         * @param choices the tag
-         */
-        public MaterialChoice(Tag<Material> choices) {
-            this(new ArrayList<>(java.util.Objects.requireNonNull(choices, "Cannot create a material choice with null tag").getValues())); // Paper - delegate to list ctor to make sure all checks are called
-        }
-
-        public MaterialChoice(List<Material> choices) {
-            Preconditions.checkArgument(choices != null, "choices");
-            Preconditions.checkArgument(!choices.isEmpty(), "Must have at least one choice");
-
-            this.choices = new ArrayList<>(choices.size());
-
-            for (Material choice : choices) {
-                Preconditions.checkArgument(choice != null, "Cannot have null choice");
-
-                if (choice.isLegacy()) {
-                    choice = Bukkit.getUnsafe().fromLegacy(new MaterialData(choice, (byte) 0), true);
-                }
-
-                Preconditions.checkArgument(!choice.isAir(), "Cannot have empty/air choice");
-                Preconditions.checkArgument(choice.isItem(), "Cannot have non-item choice %s", choice); // Paper - validate material choice input to items
-                this.choices.add(choice);
-            }
-        }
-
-        @Override
-        public boolean test(ItemStack t) {
-            for (Material match : choices) {
-                if (t.getType() == match) {
-                    return true;
-                }
-            }
-
-            return false;
-        }
-
-        @Override
-        @Deprecated(since = "1.13.1")
-        public ItemStack getItemStack() {
-            ItemStack stack = new ItemStack(choices.get(0));
-
-            // For compat
-            if (choices.size() > 1) {
-                stack.setDurability(Short.MAX_VALUE);
-            }
-
-            return stack;
-        }
-
-        public List<Material> getChoices() {
-            return Collections.unmodifiableList(choices);
-        }
-
-        @Override
-        public MaterialChoice clone() {
-            try {
-                MaterialChoice clone = (MaterialChoice) super.clone();
-                clone.choices = new ArrayList<>(choices);
-                return clone;
-            } catch (CloneNotSupportedException ex) {
-                throw new AssertionError(ex);
-            }
-        }
-
-        @Override
-        public int hashCode() {
-            int hash = 3;
-            hash = 37 * hash + Objects.hashCode(this.choices);
-            return hash;
-        }
-
-        @Override
-        public boolean equals(Object obj) {
-            if (this == obj) {
-                return true;
-            }
-            if (obj == null) {
-                return false;
-            }
-            if (getClass() != obj.getClass()) {
-                return false;
-            }
-            final MaterialChoice other = (MaterialChoice) obj;
-            if (!Objects.equals(this.choices, other.choices)) {
-                return false;
-            }
-            return true;
-        }
-
-        @Override
-        public String toString() {
-            return "MaterialChoice{" + "choices=" + choices + '}';
-        }
-
-        // Paper start - check valid ingredients
-        @Override
-        public RecipeChoice validate(final boolean allowEmptyRecipes) {
-            if (this.choices.stream().anyMatch(Material::isAir)) {
-                throw new IllegalArgumentException("RecipeChoice.MaterialChoice cannot contain air");
-            }
-            return this;
-        }
-        // Paper end - check valid ingredients
-    }
-
-    /**
      * Represents a choice that will be valid only if one of the stacks is
      * exactly matched (aside from stack size).
      */
@@ -300,7 +_,7 @@
             Preconditions.checkArgument(!choices.isEmpty(), "Must have at least one choice");
             for (ItemStack choice : choices) {
                 Preconditions.checkArgument(choice != null, "Cannot have null choice");
-                Preconditions.checkArgument(!choice.getType().isAir(), "Cannot have empty/air choice");
+                Preconditions.checkArgument(choice.getType() != ItemType.AIR, "Cannot have empty/air choice");
             }
 
             this.choices = new ArrayList<>(choices);
@@ -376,7 +_,7 @@
         // Paper start - check valid ingredients
         @Override
         public RecipeChoice validate(final boolean allowEmptyRecipes) {
-            if (this.choices.stream().anyMatch(s -> s.getType().isAir())) {
+            if (this.choices.stream().anyMatch(s -> s.getType() == ItemType.AIR)) {
                 throw new IllegalArgumentException("RecipeChoice.ExactChoice cannot contain air");
             }
             return this;
