--- a/src/main/java/org/bukkit/inventory/StonecuttingRecipe.java
+++ b/src/main/java/org/bukkit/inventory/StonecuttingRecipe.java
@@ -1,9 +_,7 @@
 package org.bukkit.inventory;
 
 import com.google.common.base.Preconditions;
-import java.util.Collections;
 import org.bukkit.Keyed;
-import org.bukkit.Material;
 import org.bukkit.NamespacedKey;
 import org.jetbrains.annotations.NotNull;
 
@@ -22,8 +_,8 @@
      * @param result The item you want the recipe to create.
      * @param source The input material.
      */
-    public StonecuttingRecipe(@NotNull NamespacedKey key, @NotNull ItemStack result, @NotNull Material source) {
-        this(key, result, new RecipeChoice.MaterialChoice(Collections.singletonList(source)));
+    public StonecuttingRecipe(@NotNull NamespacedKey key, @NotNull ItemStack result, @NotNull ItemType source) {
+        this(key, result, RecipeChoice.itemType(source));
     }
 
     /**
@@ -47,8 +_,8 @@
      * @return The changed recipe, so you can chain calls.
      */
     @NotNull
-    public StonecuttingRecipe setInput(@NotNull Material input) {
-        this.ingredient = new RecipeChoice.MaterialChoice(Collections.singletonList(input));
+    public StonecuttingRecipe setInput(@NotNull ItemType input) {
+        this.ingredient = RecipeChoice.itemType(input);
         return this;
     }
 
