From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Wed, 11 Dec 2024 19:23:25 -0700
Subject: [PATCH] 'feature patch'


diff --git a/net/minecraft/DefaultUncaughtExceptionHandler.java b/net/minecraft/DefaultUncaughtExceptionHandler.java
index adea07c8940e8ffa1ff53340df7bbbb1a2b31b05..14d54d200064d965660e044346483af04d0189f6 100644
--- a/net/minecraft/DefaultUncaughtExceptionHandler.java
+++ b/net/minecraft/DefaultUncaughtExceptionHandler.java
@@ -12,6 +12,7 @@ public class DefaultUncaughtExceptionHandler implements UncaughtExceptionHandler
 
     @Override
     public void uncaughtException(final Thread t, final Throwable e) {
+        System.out.println("im a feature");
         this.logger.error("Caught previously unhandled exception :", e);
     }
 }
