--- a/src/test/java/org/bukkit/support/DummyServerHelper.java
+++ b/src/test/java/org/bukkit/support/DummyServerHelper.java
@@ -9,8 +_,8 @@
 import net.minecraft.tags.TagKey;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.level.block.Block;
-import org.bukkit.Material;
 import org.bukkit.Server;
+import org.bukkit.block.BlockType;
 import org.bukkit.craftbukkit.CraftLootTable;
 import org.bukkit.craftbukkit.block.data.CraftBlockData;
 import org.bukkit.craftbukkit.inventory.CraftItemFactory;
@@ -40,7 +_,7 @@
 
         when(instance.getUnsafe()).then(mock -> CraftMagicNumbers.INSTANCE);
 
-        when(instance.createBlockData(any(Material.class))).then(mock -> CraftBlockData.fromString(((Material) mock.getArgument(0)).asBlockType(), null));
+        when(instance.createBlockData(any(BlockType.class))).then(mock -> CraftBlockData.fromString(((BlockType) mock.getArgument(0)), null));
 
         when(instance.getLootTable(any())).then(mock -> new CraftLootTable(mock.getArgument(0),
                 RegistryHelper.context().datapack().fullRegistries().getLootTable(ResourceKey.create(Registries.LOOT_TABLE, CraftNamespacedKey.toMinecraft(mock.getArgument(0))))));
@@ -52,14 +_,14 @@
 
             switch (registry) {
                 case org.bukkit.Tag.REGISTRY_BLOCKS -> {
-                    Preconditions.checkArgument(clazz == Material.class, "Block namespace must have block type");
+                    Preconditions.checkArgument(clazz == org.bukkit.block.BlockType.class, "Block namespace (%s) must have block type", clazz.getName());
                     TagKey<Block> blockTagKey = TagKey.create(Registries.BLOCK, key);
                     if (BuiltInRegistries.BLOCK.get(blockTagKey).isPresent()) {
                         return new CraftBlockTag(BuiltInRegistries.BLOCK, blockTagKey);
                     }
                 }
                 case org.bukkit.Tag.REGISTRY_ITEMS -> {
-                    Preconditions.checkArgument(clazz == Material.class, "Item namespace must have item type");
+                    Preconditions.checkArgument(clazz == org.bukkit.inventory.ItemType.class, "Item namespace (%s) must have item type", clazz.getName());
                     TagKey<Item> itemTagKey = TagKey.create(Registries.ITEM, key);
                     if (BuiltInRegistries.ITEM.get(itemTagKey).isPresent()) {
                         return new CraftItemTag(BuiltInRegistries.ITEM, itemTagKey);
