--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -85,12 +_,12 @@
 import net.minecraft.world.phys.Vec3;
 import org.bukkit.Bukkit;
 import org.bukkit.Location;
-import org.bukkit.Material;
 import org.bukkit.PortalType;
 import org.bukkit.Statistic.Type;
 import org.bukkit.block.Block;
 import org.bukkit.block.BlockFace;
 import org.bukkit.block.BlockState;
+import org.bukkit.block.BlockType;
 import org.bukkit.block.Sign;
 import org.bukkit.block.sign.Side;
 import org.bukkit.command.CommandSender;
@@ -277,6 +_,7 @@
 import org.bukkit.inventory.CraftingRecipe;
 import org.bukkit.inventory.EquipmentSlot;
 import org.bukkit.inventory.InventoryView;
+import org.bukkit.inventory.ItemType;
 import org.bukkit.inventory.Recipe;
 import org.bukkit.inventory.meta.BookMeta;
 import org.bukkit.inventory.view.AnvilView;
@@ -585,8 +_,8 @@
 
     private static PlayerEvent getPlayerBucketEvent(boolean isFilling, Level level, net.minecraft.world.entity.player.Player player, BlockPos changed, BlockPos clicked, Direction clickedFace, ItemStack bucket, net.minecraft.world.item.Item item, InteractionHand hand) {
         Player cplayer = (Player) player.getBukkitEntity();
-        org.bukkit.inventory.ItemStack itemInHand = org.bukkit.inventory.ItemStack.of(CraftItemType.minecraftToBukkit(item));
-        Material bucketItem = CraftItemType.minecraftToBukkit(bucket.getItem());
+        org.bukkit.inventory.ItemStack itemInHand = org.bukkit.inventory.ItemStack.of(CraftItemType.minecraftToBukkitNew(item));
+        ItemType bucketItem = CraftItemType.minecraftToBukkitNew(bucket.getItem());
 
         Block block = CraftBlock.at(level, changed);
         Block clickedBlock = CraftBlock.at(level, clicked);
@@ -960,7 +_,7 @@
 
     private static final java.util.function.Function<org.bukkit.inventory.ItemStack, Entity.DefaultDrop> FROM_FUNCTION = stack -> {
         if (stack == null) return null;
-        return new Entity.DefaultDrop(CraftItemType.bukkitToMinecraft(stack.getType()), stack, null);
+        return new Entity.DefaultDrop(CraftItemType.bukkitToMinecraftNew(stack.getType()), stack, null);
     };
 
     public static boolean callEntityDeathEvent(ServerLevel level, net.minecraft.world.entity.LivingEntity victim, DamageSource damageSource) {
@@ -1546,16 +_,13 @@
     public static BlockIgniteEvent callBlockIgniteEvent(Level level, BlockPos pos, BlockPos sourcePos) {
         Block igniter = CraftBlock.at(level, sourcePos);
         final IgniteCause cause;
-        switch (igniter.getType()) {
-            case LAVA:
-                cause = IgniteCause.LAVA;
-                break;
-            case DISPENSER:
-                cause = IgniteCause.FLINT_AND_STEEL;
-                break;
-            case FIRE: // Fire or any other unknown block counts as SPREAD.
-            default:
-                cause = IgniteCause.SPREAD;
+        final BlockType igniterType = igniter.getType();
+        if (igniterType == BlockType.LAVA) {
+            cause = IgniteCause.LAVA;
+        } else if (igniterType == BlockType.DISPENSER) {
+            cause = IgniteCause.FLINT_AND_STEEL;
+        } else { // Fire or any other unknown block counts as SPREAD.
+            cause = IgniteCause.SPREAD;
         }
 
         BlockIgniteEvent event = new BlockIgniteEvent(CraftBlock.at(level, pos), cause, igniter);
@@ -1746,9 +_,12 @@
         } else if (stat.getType() == Type.ENTITY) {
             EntityType entityType = CraftStatistic.getEntityTypeFromStatistic((net.minecraft.stats.Stat<net.minecraft.world.entity.EntityType<?>>) statistic);
             event = new PlayerStatisticIncrementEvent(player, stat, current, newValue, entityType);
+        } else if (stat.getType() == Type.ITEM) {
+            ItemType itemType = CraftStatistic.getItemTypeFromStatistic((net.minecraft.stats.Stat<net.minecraft.world.item.Item>) statistic);
+            event = new PlayerStatisticIncrementEvent(player, stat, current, newValue, itemType);
         } else {
-            Material material = CraftStatistic.getMaterialFromStatistic(statistic);
-            event = new PlayerStatisticIncrementEvent(player, stat, current, newValue, material);
+            BlockType blockType = CraftStatistic.getBlockTypeFromStatistic((net.minecraft.stats.Stat<net.minecraft.world.level.block.Block>) statistic);
+            event = new PlayerStatisticIncrementEvent(player, stat, current, newValue, blockType);
         }
         event.callEvent();
         return (Cancellable) event;
