--- a/src/main/java/org/bukkit/block/Block.java
+++ b/src/main/java/org/bukkit/block/Block.java
@@ -4,7 +_,6 @@
 import org.bukkit.Chunk;
 import org.bukkit.FluidCollisionMode;
 import org.bukkit.Location;
-import org.bukkit.Material;
 import org.bukkit.Translatable;
 import org.bukkit.World;
 import org.bukkit.block.data.Bisected;
@@ -35,15 +_,6 @@
 public interface Block extends Metadatable, Translatable, net.kyori.adventure.translation.Translatable { // Paper - translatable
 
     /**
-     * Gets the metadata for this block
-     *
-     * @return block specific metadata
-     * @deprecated Magic value
-     */
-    @Deprecated(since = "1.6.2", forRemoval = true)
-    byte getData();
-
-    /**
      * Gets the complete block data for this block
      *
      * @return block specific data
@@ -99,7 +_,7 @@
      * @return block type
      */
     @NotNull
-    Material getType();
+    BlockType getType();
 
     /**
      * Gets the light level between 0-15
@@ -305,7 +_,7 @@
      *
      * @param type Material to change this block to
      */
-    void setType(@NotNull Material type);
+    void setType(@NotNull BlockType type);
 
     /**
      * Sets the type of this block
@@ -326,7 +_,7 @@
      * @param type Material to change this block to
      * @param applyPhysics False to cancel physics on the changed block.
      */
-    void setType(@NotNull Material type, boolean applyPhysics);
+    void setType(@NotNull BlockType type, boolean applyPhysics);
 
     /**
      * Gets the face relation of this block compared to the given block.
@@ -453,7 +_,7 @@
      * Checks if this block is empty.
      * <p>
      * A block is considered empty when {@link #getType()} returns {@link
-     * Material#AIR}.
+     * BlockType#AIR}.
      *
      * @return true if this block is empty
      */
@@ -463,7 +_,7 @@
      * Checks if this block is liquid.
      * <p>
      * A block is considered liquid when {@link #getType()} returns {@link
-     * Material#WATER} or {@link Material#LAVA}.
+     * BlockType#WATER} or {@link BlockType#LAVA}.
      *
      * @return true if this block is liquid
      */
