--- a/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java
@@ -24,11 +_,11 @@
 import net.minecraft.world.level.chunk.ChunkGenerator;
 import net.minecraft.world.level.levelgen.feature.Feature;
 import org.bukkit.Location;
-import org.bukkit.Material;
 import org.bukkit.RegionAccessor;
 import org.bukkit.TreeType;
 import org.bukkit.block.Biome;
 import org.bukkit.block.BlockState;
+import org.bukkit.block.BlockType;
 import org.bukkit.block.data.BlockData;
 import org.bukkit.craftbukkit.block.CraftBiome;
 import org.bukkit.craftbukkit.block.CraftBlock;
@@ -104,8 +_,8 @@
     }
 
     @Override
-    public Material getType(int x, int y, int z) {
-        return CraftBlockType.minecraftToBukkit(this.getData(x, y, z).getBlock());
+    public BlockType getType(int x, int y, int z) {
+        return CraftBlockType.minecraftToBukkitNew(this.getData(x, y, z).getBlock());
     }
 
     private net.minecraft.world.level.block.state.BlockState getData(int x, int y, int z) {
