--- a/src/main/java/org/bukkit/craftbukkit/block/CraftDecoratedPot.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftDecoratedPot.java
@@ -10,13 +_,13 @@
 import net.minecraft.world.level.block.entity.DecoratedPotBlockEntity;
 import net.minecraft.world.level.block.entity.PotDecorations;
 import org.bukkit.Location;
-import org.bukkit.Material;
 import org.bukkit.Tag;
 import org.bukkit.World;
 import org.bukkit.block.DecoratedPot;
 import org.bukkit.craftbukkit.inventory.CraftInventoryDecoratedPot;
 import org.bukkit.craftbukkit.inventory.CraftItemType;
 import org.bukkit.inventory.DecoratedPotInventory;
+import org.bukkit.inventory.ItemType;
 import org.jetbrains.annotations.NotNull;
 
 public class CraftDecoratedPot extends CraftBlockEntityState<DecoratedPotBlockEntity> implements DecoratedPot {
@@ -69,15 +_,15 @@
     }
 
     @Override
-    public void setSherd(Side side, Material sherd) {
+    public void setSherd(Side side, ItemType sherd) {
         Preconditions.checkArgument(side != null, "side must not be null");
-        Preconditions.checkArgument(sherd == null || sherd == Material.BRICK || Tag.ITEMS_DECORATED_POT_SHERDS.isTagged(sherd), "sherd is not a valid sherd material: %s", sherd);
+        Preconditions.checkArgument(sherd == null || sherd == ItemType.BRICK || Tag.ITEMS_DECORATED_POT_SHERDS.isTagged(sherd), "sherd is not a valid sherd material: %s", sherd);
 
-        if (sherd == Material.BRICK) {
+        if (sherd == ItemType.BRICK) {
             sherd = null;
         }
 
-        final Optional<ItemStackTemplate> sherdTemplate = Optional.ofNullable(sherd).map(s -> new ItemStackTemplate(CraftItemType.bukkitToMinecraft(s)));
+        final Optional<ItemStackTemplate> sherdTemplate = Optional.ofNullable(sherd).map(s -> new ItemStackTemplate(CraftItemType.bukkitToMinecraftNew(s)));
         final PotDecorations decorations = this.getSnapshot().getDecorations();
 
         this.getSnapshot().decorations = switch (side) {
@@ -89,7 +_,7 @@
     }
 
     @Override
-    public Material getSherd(Side side) {
+    public ItemType getSherd(Side side) {
         Preconditions.checkArgument(side != null, "side must not be null");
 
         PotDecorations decorations = this.getSnapshot().getDecorations();
@@ -101,25 +_,25 @@
         };
 
         return sherdTemplate
-            .map(template -> CraftItemType.minecraftToBukkit(template.item().value()))
-            .orElse(Material.BRICK);
+            .map(template -> CraftItemType.minecraftToBukkitNew(template.item().value()))
+            .orElse(ItemType.BRICK);
     }
 
     @Override
-    public Map<Side, Material> getSherds() {
+    public Map<Side, ItemType> getSherds() {
         PotDecorations decorations = this.getSnapshot().getDecorations();
-        Function<ItemStackTemplate, Material> toMaterial = template -> CraftItemType.minecraftToBukkit(template.item().value());
+        Function<ItemStackTemplate, ItemType> toMaterial = template -> CraftItemType.minecraftToBukkitNew(template.item().value());
 
-        Map<Side, Material> sherds = new EnumMap<>(Side.class);
-        sherds.put(Side.BACK, decorations.back().map(toMaterial).orElse(Material.BRICK));
-        sherds.put(Side.LEFT, decorations.left().map(toMaterial).orElse(Material.BRICK));
-        sherds.put(Side.RIGHT, decorations.right().map(toMaterial).orElse(Material.BRICK));
-        sherds.put(Side.FRONT, decorations.front().map(toMaterial).orElse(Material.BRICK));
+        Map<Side, ItemType> sherds = new EnumMap<>(Side.class);
+        sherds.put(Side.BACK, decorations.back().map(toMaterial).orElse(ItemType.BRICK));
+        sherds.put(Side.LEFT, decorations.left().map(toMaterial).orElse(ItemType.BRICK));
+        sherds.put(Side.RIGHT, decorations.right().map(toMaterial).orElse(ItemType.BRICK));
+        sherds.put(Side.FRONT, decorations.front().map(toMaterial).orElse(ItemType.BRICK));
         return sherds;
     }
 
     @Override
-    public List<Material> getShards() {
+    public List<ItemType> getShards() {
         return List.copyOf(this.getSherds().values());
     }
 
