--- a/src/main/java/org/bukkit/craftbukkit/block/CraftDropper.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftDropper.java
@@ -4,9 +_,9 @@
 import net.minecraft.world.level.block.DropperBlock;
 import net.minecraft.world.level.block.entity.DropperBlockEntity;
 import org.bukkit.Location;
-import org.bukkit.Material;
 import org.bukkit.World;
 import org.bukkit.block.Block;
+import org.bukkit.block.BlockType;
 import org.bukkit.block.Dropper;
 import org.bukkit.craftbukkit.CraftWorld;
 import org.bukkit.craftbukkit.inventory.CraftInventory;
@@ -40,7 +_,7 @@
     public void drop() {
         this.ensureNoWorldGeneration();
         Block block = this.getBlock();
-        if (block.getType() == Material.DROPPER) {
+        if (block.getType() == BlockType.DROPPER) {
             CraftWorld world = (CraftWorld) this.getWorld();
 
             ((DropperBlock) Blocks.DROPPER).dispenseFrom(world.getHandle(), this.getHandle(), this.getPosition());
