--- a/src/main/java/org/bukkit/craftbukkit/block/CraftSkull.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftSkull.java
@@ -21,6 +_,7 @@
 import org.bukkit.SkullType;
 import org.bukkit.World;
 import org.bukkit.block.BlockFace;
+import org.bukkit.block.BlockType;
 import org.bukkit.block.Skull;
 import org.bukkit.block.data.BlockData;
 import org.bukkit.block.data.Directional;
@@ -30,6 +_,8 @@
 import org.bukkit.craftbukkit.util.CraftNamespacedKey;
 import org.bukkit.profile.PlayerProfile;
 import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
 import java.util.Objects;
 
 public class CraftSkull extends CraftBlockEntityState<SkullBlockEntity> implements Skull {
@@ -195,31 +_,26 @@
 
     @Override
     public SkullType getSkullType() {
-        switch (this.getType()) {
-            case SKELETON_SKULL:
-            case SKELETON_WALL_SKULL:
-                return SkullType.SKELETON;
-            case WITHER_SKELETON_SKULL:
-            case WITHER_SKELETON_WALL_SKULL:
-                return SkullType.WITHER;
-            case ZOMBIE_HEAD:
-            case ZOMBIE_WALL_HEAD:
-                return SkullType.ZOMBIE;
-            case PIGLIN_HEAD:
-            case PIGLIN_WALL_HEAD:
-                return SkullType.PIGLIN;
-            case PLAYER_HEAD:
-            case PLAYER_WALL_HEAD:
-                return SkullType.PLAYER;
-            case CREEPER_HEAD:
-            case CREEPER_WALL_HEAD:
-                return SkullType.CREEPER;
-            case DRAGON_HEAD:
-            case DRAGON_WALL_HEAD:
-                return SkullType.DRAGON;
-            default:
-                throw new IllegalArgumentException("Unknown SkullType for " + this.getType());
+        Map<BlockType, SkullType> typeMap = Map.ofEntries(
+            Map.entry(BlockType.SKELETON_SKULL, SkullType.SKELETON),
+            Map.entry(BlockType.SKELETON_WALL_SKULL, SkullType.SKELETON),
+            Map.entry(BlockType.WITHER_SKELETON_SKULL, SkullType.WITHER),
+            Map.entry(BlockType.WITHER_SKELETON_WALL_SKULL, SkullType.WITHER),
+            Map.entry(BlockType.ZOMBIE_HEAD, SkullType.ZOMBIE),
+            Map.entry(BlockType.ZOMBIE_WALL_HEAD, SkullType.ZOMBIE),
+            Map.entry(BlockType.PIGLIN_HEAD, SkullType.PIGLIN),
+            Map.entry(BlockType.PIGLIN_WALL_HEAD, SkullType.PIGLIN),
+            Map.entry(BlockType.PLAYER_HEAD, SkullType.PLAYER),
+            Map.entry(BlockType.PLAYER_WALL_HEAD, SkullType.PLAYER),
+            Map.entry(BlockType.CREEPER_HEAD, SkullType.CREEPER),
+            Map.entry(BlockType.CREEPER_WALL_HEAD, SkullType.CREEPER),
+            Map.entry(BlockType.DRAGON_HEAD, SkullType.DRAGON),
+            Map.entry(BlockType.DRAGON_WALL_HEAD, SkullType.DRAGON)
+        );
+        if (!typeMap.containsKey(this.getType())) {
+            throw new IllegalArgumentException("Unknown SkullType for " + this.getType());
         }
+        return typeMap.get(this.getType());
     }
 
     @Override
