--- a/net/minecraft/world/item/ServerItemCooldowns.java
+++ b/net/minecraft/world/item/ServerItemCooldowns.java
@@ -17,7 +_,7 @@
         final Identifier cooldownGroup = this.getCooldownGroup(item);
         final io.papermc.paper.event.player.PlayerItemCooldownEvent event = new io.papermc.paper.event.player.PlayerItemCooldownEvent(
             this.player.getBukkitEntity(),
-            org.bukkit.craftbukkit.inventory.CraftItemType.minecraftToBukkit(item.getItem()),
+            org.bukkit.craftbukkit.inventory.CraftItemType.minecraftToBukkitNew(item.getItem()),
             org.bukkit.craftbukkit.util.CraftNamespacedKey.fromMinecraft(cooldownGroup),
             duration
         );
