--- a/src/main/java/org/bukkit/event/block/BlockPhysicsEvent.java
+++ b/src/main/java/org/bukkit/event/block/BlockPhysicsEvent.java
@@ -1,7 +_,7 @@
 package org.bukkit.event.block;
 
-import org.bukkit.Material;
 import org.bukkit.block.Block;
+import org.bukkit.block.BlockType;
 import org.bukkit.block.data.BlockData;
 import org.bukkit.event.Cancellable;
 import org.bukkit.event.HandlerList;
@@ -72,7 +_,7 @@
      * @return Changed block's type
      */
     @NotNull
-    public Material getChangedType() {
+    public BlockType getChangedType() {
         return this.changed.getMaterial();
     }
 
