--- a/src/main/java/org/bukkit/event/block/BlockCanBuildEvent.java
+++ b/src/main/java/org/bukkit/event/block/BlockCanBuildEvent.java
@@ -1,7 +_,7 @@
 package org.bukkit.event.block;
 
-import org.bukkit.Material;
 import org.bukkit.block.Block;
+import org.bukkit.block.BlockType;
 import org.bukkit.block.data.BlockData;
 import org.bukkit.entity.Player;
 import org.bukkit.event.HandlerList;
@@ -68,7 +_,7 @@
      * @return The Material that we are trying to place
      */
     @NotNull
-    public Material getMaterial() {
+    public BlockType getMaterial() {
         return this.blockData.getMaterial();
     }
 
