--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -18,6 +_,7 @@
 import net.kyori.adventure.text.Component;
 import org.bukkit.Warning.WarningState;
 import org.bukkit.advancement.Advancement;
+import org.bukkit.block.BlockType;
 import org.bukkit.block.data.BlockData;
 import org.bukkit.boss.BarColor;
 import org.bukkit.boss.BarFlag;
@@ -1262,7 +_,7 @@
      * @param world The world the crafting takes place in.
      * @param player The player to imitate the crafting event on.
      * @return the {@link ItemStack} resulting from the given crafting matrix, if no recipe is found
-     * an ItemStack of {@link Material#AIR} is returned.
+     * an ItemStack of {@link org.bukkit.inventory.ItemType#AIR} is returned.
      */
     @NotNull
     public static ItemStack craftItem(@NotNull ItemStack @NotNull [] craftingMatrix, @NotNull World world, @NotNull Player player) {
@@ -1285,7 +_,7 @@
      *                       Must not contain more than 9 items.
      * @param world The world the crafting takes place in.
      * @return the {@link ItemStack} resulting from the given crafting matrix, if no recipe is found
-     * an ItemStack of {@link Material#AIR} is returned.
+     * an ItemStack of {@link org.bukkit.inventory.ItemType#AIR} is returned.
      */
     @NotNull
     public static ItemStack craftItem(@NotNull ItemStack @NotNull [] craftingMatrix, @NotNull World world) {
@@ -2482,19 +_,19 @@
     }
 
     /**
-     * Creates a new {@link BlockData} instance for the specified Material, with
+     * Creates a new {@link BlockData} instance for the specified BlockType, with
      * all properties initialized to unspecified defaults.
      *
      * @param material the material
      * @return new data instance
      */
     @NotNull
-    public static BlockData createBlockData(@NotNull Material material) {
+    public static BlockData createBlockData(@NotNull BlockType material) {
         return server.createBlockData(material);
     }
 
     /**
-     * Creates a new {@link BlockData} instance for the specified Material, with
+     * Creates a new {@link BlockData} instance for the specified BlockType, with
      * all properties initialized to unspecified defaults.
      *
      * @param material the material
@@ -2502,7 +_,7 @@
      * @return new data instance
      */
     @NotNull
-    public static BlockData createBlockData(@NotNull Material material, @Nullable Consumer<? super BlockData> consumer) {
+    public static BlockData createBlockData(@NotNull BlockType material, @Nullable Consumer<? super BlockData> consumer) {
         return server.createBlockData(material, consumer);
     }
 
@@ -2520,7 +_,7 @@
     }
 
     /**
-     * Creates a new {@link BlockData} instance for the specified Material, with
+     * Creates a new {@link BlockData} instance for the specified BlockType, with
      * all properties initialized to unspecified defaults, except for those
      * provided in data.
      *
@@ -2531,7 +_,7 @@
      */
     @NotNull
     @Contract("null, null -> fail")
-    public static BlockData createBlockData(@Nullable Material material, @Nullable String data) throws IllegalArgumentException {
+    public static BlockData createBlockData(@Nullable BlockType material, @Nullable String data) throws IllegalArgumentException {
         return server.createBlockData(material, data);
     }
 
