--- a/src/main/java/org/bukkit/Registry.java
+++ b/src/main/java/org/bukkit/Registry.java
@@ -187,12 +_,6 @@
      */
     Registry<LootTables> LOOT_TABLES = new SimpleRegistry<>(LootTables.class);
     /**
-     * Server materials.
-     *
-     * @see Material
-     */
-    Registry<Material> MATERIAL = new SimpleRegistry<>(Material.class, (mat) -> !mat.isLegacy());
-    /**
      * Server menus.
      *
      * @see MenuType
