--- a/src/main/java/org/bukkit/UnsafeValues.java
+++ b/src/main/java/org/bukkit/UnsafeValues.java
@@ -7,11 +_,9 @@
 import net.kyori.adventure.key.Key;
 import net.kyori.adventure.text.event.HoverEvent;
 import org.bukkit.advancement.Advancement;
-import org.bukkit.block.data.BlockData;
 import org.bukkit.entity.Entity;
 import org.bukkit.event.entity.CreatureSpawnEvent;
 import org.bukkit.inventory.ItemStack;
-import org.bukkit.material.MaterialData;
 import org.bukkit.plugin.InvalidPluginException;
 import org.bukkit.plugin.PluginDescriptionFile;
 import org.bukkit.potion.PotionType;
@@ -30,18 +_,6 @@
  */
 @Deprecated(since = "1.7.2")
 public interface UnsafeValues {
-
-    Material toLegacy(Material material);
-
-    Material fromLegacy(Material material);
-
-    Material fromLegacy(MaterialData material);
-
-    Material fromLegacy(MaterialData material, boolean itemPriority);
-
-    BlockData fromLegacy(Material material, byte data);
-
-    Material getMaterial(String material, int version);
 
     int getDataVersion();
 
