--- a/src/main/java/org/bukkit/inventory/meta/FireworkEffectMeta.java
+++ b/src/main/java/org/bukkit/inventory/meta/FireworkEffectMeta.java
@@ -1,13 +_,12 @@
 package org.bukkit.inventory.meta;
 
 import org.bukkit.FireworkEffect;
-import org.bukkit.Material;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 /**
  * Represents a meta that can store a single FireworkEffect. An example
- * includes {@link Material#FIREWORK_STAR}.
+ * includes {@link org.bukkit.inventory.ItemType#FIREWORK_STAR}.
  */
 public interface FireworkEffectMeta extends ItemMeta {
 
