--- a/src/main/java/org/bukkit/craftbukkit/inventory/components/CraftToolComponent.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/components/CraftToolComponent.java
@@ -15,8 +_,8 @@
 import net.minecraft.core.registries.Registries;
 import net.minecraft.world.item.component.Tool;
 import net.minecraft.world.level.block.Block;
-import org.bukkit.Material;
 import org.bukkit.Tag;
+import org.bukkit.block.BlockType;
 import org.bukkit.configuration.serialization.SerializableAs;
 import org.bukkit.craftbukkit.block.CraftBlockType;
 import org.bukkit.craftbukkit.configuration.ConfigSerializationUtil;
@@ -105,34 +_,32 @@
     }
 
     @Override
-    public ToolRule addRule(Material block, Float speed, Boolean correctForDrops) {
+    public ToolRule addRule(BlockType block, Float speed, Boolean correctForDrops) {
         Preconditions.checkArgument(block != null, "block must not be null");
-        Preconditions.checkArgument(block.isBlock(), "block must be a block type, given %s", block.getKey());
         if (speed != null) {
             requirePositive(speed, "speed");
         }
 
-        Holder.Reference<Block> nmsBlock = CraftBlockType.bukkitToMinecraft(block).builtInRegistryHolder();
+        Holder.Reference<Block> nmsBlock = CraftBlockType.bukkitToMinecraftNew(block).builtInRegistryHolder();
         return this.addRule(HolderSet.direct(nmsBlock), speed, correctForDrops);
     }
 
     @Override
-    public ToolRule addRule(Collection<Material> blocks, Float speed, Boolean correctForDrops) {
+    public ToolRule addRule(Collection<BlockType> blocks, Float speed, Boolean correctForDrops) {
         if (speed != null) {
             requirePositive(speed, "speed");
         }
         List<Holder.Reference<Block>> nmsBlocks = new ArrayList<>(blocks.size());
 
-        for (Material material : blocks) {
-            Preconditions.checkArgument(material.isBlock(), "blocks contains non-block type: %s", material.getKey());
-            nmsBlocks.add(CraftBlockType.bukkitToMinecraft(material).builtInRegistryHolder());
+        for (BlockType material : blocks) {
+            nmsBlocks.add(CraftBlockType.bukkitToMinecraftNew(material).builtInRegistryHolder());
         }
 
         return this.addRule(HolderSet.direct(nmsBlocks), speed, correctForDrops);
     }
 
     @Override
-    public ToolRule addRule(Tag<Material> tag, Float speed, Boolean correctForDrops) {
+    public ToolRule addRule(Tag<BlockType> tag, Float speed, Boolean correctForDrops) {
         Preconditions.checkArgument(tag instanceof CraftBlockTag, "tag must be a block tag");
         return this.addRule(((CraftBlockTag) tag).getHandle(), speed == null ? null : requirePositive(speed, "speed"), correctForDrops);
     }
@@ -229,29 +_,24 @@
         }
 
         @Override
-        public Collection<Material> getBlocks() {
-            return this.handle.blocks().stream().map(Holder::value).map(CraftBlockType::minecraftToBukkit).collect(Collectors.toList());
+        public Collection<BlockType> getBlocks() {
+            return this.handle.blocks().stream().map(Holder::value).map(CraftBlockType::minecraftToBukkitNew).collect(Collectors.toList());
         }
 
         @Override
-        public void setBlocks(Material block) {
+        public void setBlocks(BlockType block) {
             Preconditions.checkArgument(block != null, "block must not be null");
-            Preconditions.checkArgument(block.isBlock(), "block must be a block type, given %s", block.getKey());
-            this.handle = new Tool.Rule(HolderSet.direct(CraftBlockType.bukkitToMinecraft(block).builtInRegistryHolder()), this.handle.speed(), this.handle.correctForDrops());
+            this.handle = new Tool.Rule(HolderSet.direct(CraftBlockType.bukkitToMinecraftNew(block).builtInRegistryHolder()), this.handle.speed(), this.handle.correctForDrops());
         }
 
         @Override
-        public void setBlocks(Collection<Material> blocks) {
+        public void setBlocks(Collection<BlockType> blocks) {
             Preconditions.checkArgument(blocks != null, "blocks must not be null");
-            for (Material material : blocks) {
-                Preconditions.checkArgument(material.isBlock(), "blocks contains non-block type: %s", material.getKey());
-            }
-
-            this.handle = new Tool.Rule(HolderSet.direct((List) blocks.stream().map(CraftBlockType::bukkitToMinecraft).map(Block::builtInRegistryHolder).collect(Collectors.toList())), this.handle.speed(), this.handle.correctForDrops());
+            this.handle = new Tool.Rule(HolderSet.direct((List) blocks.stream().map(CraftBlockType::bukkitToMinecraftNew).map(Block::builtInRegistryHolder).collect(Collectors.toList())), this.handle.speed(), this.handle.correctForDrops());
         }
 
         @Override
-        public void setBlocks(Tag<Material> tag) {
+        public void setBlocks(Tag<BlockType> tag) {
             Preconditions.checkArgument(tag instanceof CraftBlockTag, "tag must be a block tag");
             this.handle = new Tool.Rule(((CraftBlockTag) tag).getHandle(), this.handle.speed(), this.handle.correctForDrops());
         }
