--- a/src/main/java/org/bukkit/generator/ChunkGenerator.java
+++ b/src/main/java/org/bukkit/generator/ChunkGenerator.java
@@ -6,12 +_,11 @@
 import org.bukkit.Bukkit;
 import org.bukkit.HeightMap;
 import org.bukkit.Location;
-import org.bukkit.Material;
 import org.bukkit.World;
 import org.bukkit.block.Biome;
 import org.bukkit.block.Block;
+import org.bukkit.block.BlockType;
 import org.bukkit.block.data.BlockData;
-import org.bukkit.material.MaterialData;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 import org.jetbrains.annotations.Range;
@@ -322,10 +_,10 @@
         case NETHER:
             return true;
         case THE_END:
-            return highest.getType() != Material.AIR && highest.getType() != Material.WATER && highest.getType() != Material.LAVA;
+            return highest.getType() != BlockType.AIR && highest.getType() != BlockType.WATER && highest.getType() != BlockType.LAVA;
         case NORMAL:
         default:
-            return highest.getType() == Material.SAND || highest.getType() == Material.GRAVEL;
+            return highest.getType() == BlockType.SAND || highest.getType() == BlockType.GRAVEL;
         }
     }
 
@@ -646,21 +_,7 @@
          * @param z the z location in the chunk from 0-15 inclusive
          * @param material the type to set the block to
          */
-        public void setBlock(int x, int y, int z, @NotNull Material material);
-
-        /**
-         * Set the block at x,y,z in the chunk data to material.
-         *
-         * Setting blocks outside the chunk's bounds does nothing.
-         *
-         * @param x the x location in the chunk from 0-15 inclusive
-         * @param y the y location in the chunk from minHeight (inclusive) - maxHeight (exclusive)
-         * @param z the z location in the chunk from 0-15 inclusive
-         * @param material the type to set the block to
-         * @deprecated use {@link #setBlock(int, int, int, BlockData)}
-         */
-        @Deprecated // Paper
-        public void setBlock(int x, int y, int z, @NotNull MaterialData material);
+        public void setBlock(int x, int y, int z, @NotNull BlockType material);
 
         /**
          * Set the block at x,y,z in the chunk data to material.
@@ -688,25 +_,7 @@
          * @param zMax maximum z location (exclusive) in the chunk to set
          * @param material the type to set the blocks to
          */
-        public void setRegion(int xMin, int yMin, int zMin, int xMax, int yMax, int zMax, @NotNull Material material);
-
-        /**
-         * Set a region of this chunk from xMin, yMin, zMin (inclusive)
-         * to xMax, yMax, zMax (exclusive) to material.
-         *
-         * Setting blocks outside the chunk's bounds does nothing.
-         *
-         * @param xMin minimum x location (inclusive) in the chunk to set
-         * @param yMin minimum y location (inclusive) in the chunk to set
-         * @param zMin minimum z location (inclusive) in the chunk to set
-         * @param xMax maximum x location (exclusive) in the chunk to set
-         * @param yMax maximum y location (exclusive) in the chunk to set
-         * @param zMax maximum z location (exclusive) in the chunk to set
-         * @param material the type to set the blocks to
-         * @deprecated use {@link #setRegion(int, int, int, int, int, int, BlockData)}
-         */
-        @Deprecated // Paper
-        public void setRegion(int xMin, int yMin, int zMin, int xMax, int yMax, int zMax, @NotNull MaterialData material);
+        public void setRegion(int xMin, int yMin, int zMin, int xMax, int yMax, int zMax, @NotNull BlockType material);
 
         /**
          * Set a region of this chunk from xMin, yMin, zMin (inclusive) to xMax,
@@ -732,25 +_,10 @@
          * @param x the x location in the chunk from 0-15 inclusive
          * @param y the y location in the chunk from minHeight (inclusive) - maxHeight (exclusive)
          * @param z the z location in the chunk from 0-15 inclusive
-         * @return the type of the block or Material.AIR if x, y or z are outside the chunk's bounds
-         */
-        @NotNull
-        public Material getType(int x, int y, int z);
-
-        /**
-         * Get the type and data of the block at x, y, z.
-         *
-         * Getting blocks outside the chunk's bounds returns air.
-         *
-         * @param x the x location in the chunk from 0-15 inclusive
-         * @param y the y location in the chunk from minHeight (inclusive) - maxHeight (exclusive)
-         * @param z the z location in the chunk from 0-15 inclusive
-         * @return the type and data of the block or the MaterialData for air if x, y or z are outside the chunk's bounds
-         * @deprecated use {@link #getBlockData(int, int, int)}
-         */
-        @NotNull
-        @Deprecated // Paper
-        public MaterialData getTypeAndData(int x, int y, int z);
+         * @return the type of the block or BlockType.AIR if x, y or z are outside the chunk's bounds
+         */
+        @NotNull
+        public BlockType getType(int x, int y, int z);
 
         /**
          * Get the type and data of the block at x, y, z.
@@ -764,20 +_,6 @@
          */
         @NotNull
         public BlockData getBlockData(int x, int y, int z);
-
-        /**
-         * Get the block data at x,y,z in the chunk data.
-         *
-         * Getting blocks outside the chunk's bounds returns 0.
-         *
-         * @param x the x location in the chunk from 0-15 inclusive
-         * @param y the y location in the chunk from minHeight (inclusive) - maxHeight (exclusive)
-         * @param z the z location in the chunk from 0-15 inclusive
-         * @return the block data value or air if x, y or z are outside the chunk's bounds
-         * @deprecated Uses magic values
-         */
-        @Deprecated(since = "1.8.8")
-        public byte getData(int x, int y, int z);
 
         /**
          * Get the current height of a position in the chunk data.
