--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftStrider.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftStrider.java
@@ -4,9 +_,9 @@
 import net.minecraft.world.entity.EquipmentSlot;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
-import org.bukkit.Material;
 import org.bukkit.craftbukkit.CraftServer;
 import org.bukkit.entity.Strider;
+import org.bukkit.inventory.ItemType;
 
 public class CraftStrider extends CraftAnimals implements Strider {
 
@@ -69,7 +_,7 @@
     }
 
     @Override
-    public Material getSteerMaterial() {
-        return Material.WARPED_FUNGUS_ON_A_STICK;
+    public ItemType getSteerMaterial() {
+        return ItemType.WARPED_FUNGUS_ON_A_STICK;
     }
 }
