--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftAnimals.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftAnimals.java
@@ -5,11 +_,11 @@
 import net.minecraft.Optionull;
 import net.minecraft.world.entity.EntityReference;
 import net.minecraft.world.entity.animal.Animal;
-import org.bukkit.Material;
 import org.bukkit.craftbukkit.CraftServer;
 import org.bukkit.craftbukkit.inventory.CraftItemStack;
 import org.bukkit.entity.Animals;
 import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.ItemType;
 
 public class CraftAnimals extends CraftAgeable implements Animals {
 
@@ -54,7 +_,7 @@
     }
 
     @Override
-    public boolean isBreedItem(Material material) {
+    public boolean isBreedItem(ItemType material) {
         return this.isBreedItem(new ItemStack(material));
     }
 }
