--- a/src/test/java/org/bukkit/craftbukkit/inventory/CompositeSerializationTest.java
+++ b/src/test/java/org/bukkit/craftbukkit/inventory/CompositeSerializationTest.java
@@ -4,11 +_,11 @@
 import java.util.Arrays;
 import java.util.List;
 import org.bukkit.ChatColor;
-import org.bukkit.Material;
 import org.bukkit.configuration.InvalidConfigurationException;
 import org.bukkit.configuration.file.YamlConfiguration;
 import org.bukkit.enchantments.Enchantment;
 import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.ItemType;
 import org.bukkit.inventory.meta.ItemMeta;
 import org.bukkit.support.environment.VanillaFeature;
 import org.junit.jupiter.api.Test;
@@ -29,26 +_,26 @@
         YamlConfiguration out = this.getConfig();
 
         List<ItemStack> stacks = new ArrayList<ItemStack>();
-        stacks.add(new ItemStack(Material.STONE));
-        stacks.add(new ItemStack(Material.SHORT_GRASS));
-        stacks.add(new ItemStack(Material.DIRT));
-        stacks.add(new ItemStack(Material.COBBLESTONE, 17));
-        stacks.add(new ItemStack(Material.OAK_PLANKS, 63));
-        stacks.add(new ItemStack(Material.OAK_SAPLING, 1, (short) 1));
-        stacks.add(new ItemStack(Material.OAK_LEAVES, 32, (short) 2));
+        stacks.add(new ItemStack(ItemType.STONE));
+        stacks.add(new ItemStack(ItemType.SHORT_GRASS));
+        stacks.add(new ItemStack(ItemType.DIRT));
+        stacks.add(new ItemStack(ItemType.COBBLESTONE, 17));
+        stacks.add(new ItemStack(ItemType.OAK_PLANKS, 63));
+        stacks.add(new ItemStack(ItemType.OAK_SAPLING, 1));
+        stacks.add(new ItemStack(ItemType.OAK_LEAVES, 32));
 
-        ItemStack item7 = new ItemStack(Material.IRON_SHOVEL);
+        ItemStack item7 = new ItemStack(ItemType.IRON_SHOVEL);
         item7.addUnsafeEnchantment(Enchantment.FIRE_PROTECTION, 1);
         stacks.add(item7);
 
-        ItemStack item8 = new ItemStack(Material.IRON_PICKAXE);
+        ItemStack item8 = new ItemStack(ItemType.IRON_PICKAXE);
         item8.addUnsafeEnchantment(Enchantment.FEATHER_FALLING, 2);
         item8.addUnsafeEnchantment(Enchantment.BLAST_PROTECTION, 1);
         item8.addUnsafeEnchantment(Enchantment.PROJECTILE_PROTECTION, 5);
         item8.addUnsafeEnchantment(Enchantment.RESPIRATION, 4);
         stacks.add(item8);
 
-        ItemStack item9 = new ItemStack(Material.APPLE);
+        ItemStack item9 = new ItemStack(ItemType.APPLE);
         ItemMeta meta9 = item9.getItemMeta();
         meta9.setDisplayName(ChatColor.RED + "DisplayName");
         meta9.setLocalizedName(ChatColor.AQUA + "LocalizedName");
@@ -70,4 +_,3 @@
         }
     }
 }
-
