--- a/src/test/java/org/bukkit/craftbukkit/inventory/DeprecatedItemMetaCustomValueTest.java
+++ b/src/test/java/org/bukkit/craftbukkit/inventory/DeprecatedItemMetaCustomValueTest.java
@@ -8,10 +_,10 @@
 import java.util.Map;
 import java.util.UUID;
 import org.bukkit.Bukkit;
-import org.bukkit.Material;
 import org.bukkit.NamespacedKey;
 import org.bukkit.configuration.file.YamlConfiguration;
 import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.ItemType;
 import org.bukkit.inventory.meta.ItemMeta;
 import org.bukkit.inventory.meta.tags.CustomItemTagContainer;
 import org.bukkit.inventory.meta.tags.ItemTagAdapterContext;
@@ -80,7 +_,7 @@
     }
 
     private ItemMeta createNewItemMeta() {
-        return Bukkit.getItemFactory().getItemMeta(Material.DIAMOND_PICKAXE);
+        return Bukkit.getItemFactory().getItemMeta(ItemType.DIAMOND_PICKAXE);
     }
 
     private NamespacedKey requestKey(String keyName) {
@@ -110,7 +_,7 @@
 
     @Test
     public void testYAMLStoring() {
-        ItemStack stack = new ItemStack(Material.DIAMOND);
+        ItemStack stack = new ItemStack(ItemType.DIAMOND);
         CraftMetaItem meta = this.createComplexItemMeta();
         stack.setItemMeta(meta);
 
@@ -121,12 +_,12 @@
         YamlConfiguration loadedConfig = YamlConfiguration.loadConfiguration(new StringReader(configValue));
 
         assertEquals(stack, loadedConfig.getSerializable("testpath", ItemStack.class));
-        assertNotEquals(new ItemStack(Material.DIAMOND), loadedConfig.getSerializable("testpath", ItemStack.class));
+        assertNotEquals(new ItemStack(ItemType.DIAMOND), loadedConfig.getSerializable("testpath", ItemStack.class));
     }
 
     @Test
     public void testCorrectType() {
-        ItemStack stack = new ItemStack(Material.DIAMOND);
+        ItemStack stack = new ItemStack(ItemType.DIAMOND);
         CraftMetaItem meta = this.createComplexItemMeta();
 
         meta.getCustomTagContainer().setCustomTag(this.requestKey("int"), ItemTagType.STRING, "1");
