--- a/src/test/java/org/bukkit/craftbukkit/inventory/ItemStackSkullTest.java
+++ b/src/test/java/org/bukkit/craftbukkit/inventory/ItemStackSkullTest.java
@@ -4,15 +_,15 @@
 import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Stream;
-import org.bukkit.Material;
 import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.ItemType;
 import org.bukkit.inventory.meta.SkullMeta;
 import org.junit.jupiter.params.provider.Arguments;
 
 public class ItemStackSkullTest extends ItemStackTest {
 
     public static Stream<Arguments> data() {
-        return ItemProvider.compound(ItemStackSkullTest.operators(), "%s %s", NAME_PARAMETER, Material.PLAYER_HEAD);
+        return ItemProvider.compound(ItemStackSkullTest.operators(), "%s %s", NAME_PARAMETER, ItemType.PLAYER_HEAD);
     }
 
     @SuppressWarnings("unchecked")
