--- a/src/test/java/org/bukkit/craftbukkit/inventory/ItemStackFireworkChargeTest.java
+++ b/src/test/java/org/bukkit/craftbukkit/inventory/ItemStackFireworkChargeTest.java
@@ -7,15 +_,15 @@
 import org.bukkit.Color;
 import org.bukkit.FireworkEffect;
 import org.bukkit.FireworkEffect.Type;
-import org.bukkit.Material;
 import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.ItemType;
 import org.bukkit.inventory.meta.FireworkEffectMeta;
 import org.junit.jupiter.params.provider.Arguments;
 
 public class ItemStackFireworkChargeTest extends ItemStackTest {
 
     public static Stream<Arguments> data() {
-        return ItemProvider.compound(ItemStackFireworkChargeTest.operators(), "%s %s", NAME_PARAMETER, Material.FIREWORK_STAR);
+        return ItemProvider.compound(ItemStackFireworkChargeTest.operators(), "%s %s", NAME_PARAMETER, ItemType.FIREWORK_STAR);
     }
 
     @SuppressWarnings("unchecked")
