--- a/src/test/java/org/bukkit/craftbukkit/inventory/ItemTypeTest.java
+++ b/src/test/java/org/bukkit/craftbukkit/inventory/ItemTypeTest.java
@@ -5,7 +_,6 @@
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import java.lang.reflect.Field;
 import java.lang.reflect.ParameterizedType;
-import org.bukkit.Material;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.inventory.ItemType;
 import org.bukkit.inventory.meta.ArmorMeta;
@@ -31,7 +_,7 @@
 
             ItemType type = (ItemType) f.get(null);
 
-            ItemMeta meta = new ItemStack(type.asMaterial()).getItemMeta();
+            ItemMeta meta = new ItemStack(type).getItemMeta();
             Class<?> internal = meta == null ? CraftMetaItem.class : meta.getClass();
             Class<?>[] interfaces = internal.getInterfaces();
             Class<?> expected;
@@ -54,8 +_,7 @@
     @Test
     public void testStaticItemTypeUsage() {
         final ItemStack itemStack = ItemType.DIAMOND.createItemStack();
-        Assertions.assertEquals(itemStack.getType(), Material.DIAMOND);
-        Assertions.assertEquals(itemStack.getType().asItemType(), ItemType.DIAMOND);
+        Assertions.assertEquals(itemStack.getType(), ItemType.DIAMOND);
     }
 
     @Test
