--- a/src/main/java/org/bukkit/inventory/meta/components/ToolComponent.java
+++ b/src/main/java/org/bukkit/inventory/meta/components/ToolComponent.java
@@ -2,8 +_,8 @@
 
 import java.util.Collection;
 import java.util.List;
-import org.bukkit.Material;
 import org.bukkit.Tag;
+import org.bukkit.block.BlockType;
 import org.bukkit.configuration.serialization.ConfigurationSerializable;
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
@@ -80,7 +_,7 @@
      * @return the {@link ToolRule} instance that was added to this tool
      */
     @NotNull
-    ToolRule addRule(@NotNull Material block, @Nullable Float speed, @Nullable Boolean correctForDrops);
+    ToolRule addRule(@NotNull BlockType block, @Nullable Float speed, @Nullable Boolean correctForDrops);
 
     /**
      * Add a new rule to this tool component, which provides further information
@@ -96,7 +_,7 @@
      * @return the {@link ToolRule} instance that was added to this tool
      */
     @NotNull
-    ToolRule addRule(@NotNull Collection<Material> blocks, @Nullable Float speed, @Nullable Boolean correctForDrops);
+    ToolRule addRule(@NotNull Collection<BlockType> blocks, @Nullable Float speed, @Nullable Boolean correctForDrops);
 
     /**
      * Add a new rule to this tool component, which provides further information
@@ -115,7 +_,7 @@
      * tag
      */
     @NotNull
-    ToolRule addRule(@NotNull Tag<Material> tag, @Nullable Float speed, @Nullable Boolean correctForDrops);
+    ToolRule addRule(@NotNull Tag<BlockType> tag, @Nullable Float speed, @Nullable Boolean correctForDrops);
 
     /**
      * Remove the given {@link ToolRule} from this tool.
@@ -137,21 +_,21 @@
          * @return the blocks
          */
         @NotNull
-        Collection<Material> getBlocks();
+        Collection<BlockType> getBlocks();
 
         /**
          * Set the block type to which this rule applies.
          *
          * @param block the block type
          */
-        void setBlocks(@NotNull Material block);
+        void setBlocks(@NotNull BlockType block);
 
         /**
          * Set the block types to which this rule applies.
          *
          * @param blocks the block types
          */
-        void setBlocks(@NotNull Collection<Material> blocks);
+        void setBlocks(@NotNull Collection<BlockType> blocks);
 
         /**
          * Set the block types (represented as a block {@link Tag}) to which
@@ -161,7 +_,7 @@
          * @throws IllegalArgumentException if the passed {@code tag} is not a
          * block tag
          */
-        void setBlocks(@NotNull Tag<Material> tag);
+        void setBlocks(@NotNull Tag<BlockType> tag);
 
         /**
          * Get the mining speed of this rule. If non-null, this speed value is
